### DISCUSSION TITLE
Would you recommend a custom format or a filter?

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#10789] [Q&A] [answered]
### Description

@jimjam-slam and I have been working on the [Closeread custom format](https://github.com/qmd-lab/closeread) and recently we were discussing whether to convert it from a custom format to a filter. Here are the pros and cons of the filter that we were able to identify:

**Pros**
- It would be  possible for users to create a closeread custom format.

**Cons**
- It *feels* like a categorically different format even though it isn't (like `revealjs`), so for casual users it, calling it a filter might obscure what it is.
- Passing metadata under the custom format feels more natural than triggering the filter and then having a separate key to pass metadata options.

What are y'alls thoughts on the tradeoffs here?

@georgestagg @coatless How did you think about this with `webr` (a filter) and `quarto-live` (a custom format)? As an avid user of `format: live-html`, I am bummed I can't make a custom format ...

**Comment by coatless** [comment, marked as answer]
You can make a custom format on top of an existing format, c.f.

https://github.com/r-wasm/quarto-live/issues/49

**Reply by andrewpbray** [reply]
Ah, interesting. So, the way to make a custom `live-html` format isn't to hope for `mycustom-live-html` but just make `mycustom-html` and then add the relevant yaml fields from the `quarto-live` `_extension.yml`?

Thanks!
